### PR TITLE
Release new crate versions

### PR DESCRIPTION
**Pull request by dannycjones**
## Description of change

There's not many changes here, but let's get what we have out.

Relevant issues: N/A

## Does this change impact existing behavior?

This change itself? No.

The crate releases however include one breaking change for `mountpoint-s3-crt`: "Return type of `common::Uri::host_port` changed from `u16` to `u32`".

Open question: should this be a minor rather than patch update for `mountpoint-s3-crt` version?

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3-crt/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## Unreleased
 
+## v0.5.2 (January 12, 2024)
+
 * Update to latest CRT dependencies
 * Return type of `common::Uri::host_port` changed from `u16` to `u32`
```
Comment (dannycjones) [inline mountpoint-s3-crt/CHANGELOG.md:6]: Should this be a minor version bump instead of patch version?

**Comment by sauraank** [comment]
Why do we release new versions of dependency crate like mountpoint-s3-client ? Like even before release I can pull the latest changes in them along with mountpoint crate if I use --recurse-submodules .

File: mountpoint-s3-crt/CHANGELOG.md (hunk 1 above)
Comment (passaro) [reply mountpoint-s3-crt/CHANGELOG.md:6]: It could be, but rules for unstable (0.x.y) releases are not so strict (I think).

**Review by passaro — APPROVED** [review]
LGTM

**Review by monthonk — APPROVED** [review]

File: mountpoint-s3-crt/CHANGELOG.md (hunk 1 above)
Comment (jamesbornholt) [reply mountpoint-s3-crt/CHANGELOG.md:6]: This should be a minor version bump for sure (it's API-incompatible).

**Comment by dannycjones** [comment]
Crate checks are expected to fail and are non-blocking: #658

**Review by monthonk — APPROVED** [review]
LGTM

**Pull request merged**